vllm/engine/arg_utils.py

```diff
@@ -411,7 +412,14 @@ def from_cli_args(cls, args: argparse.Namespace) -> 'EngineArgs':
         # Get the list of attributes of this dataclass.
         attrs = [attr.name for attr in dataclasses.fields(cls)]
         # Set the attributes from the parsed arguments.
-        engine_args = cls(**{attr: getattr(args, attr) for attr in attrs})
+        engine_args = cls(**{
+            attr: getattr(args, attr)
+            for attr in attrs if hasattr(args, attr)
+        })
+        if hasattr(args, "served_model_name"):
```

Comment:
Thanks for your suggestion! 

I adopted this approach as a means to adapt to `vllm.entrypoint.apiserver`. For now, `served_model_name ` only appears in `vllm.entrypoint.openai.apiserver` and currently has no description,  when using `vllm.entrypoint.apiserver`,  `served_model_name` is not an attribute of `args`.

Perhaps we could consider to complete `served_model_name` parameter in  `vllm.entrypoint.apiserver` as another approach? If so, the implementation here could be simplified as suggested. And this change will not affect anyone using `AsyncEngineArgs`/`EngineArgs ` directly.

I believe `served_model_name` is useful for all scenarios, if it can not be set, like in `vllm.entrypoint.apiserver`, sometimes a local path will be treated as the model's label, this doesn't appear to be the right way.

Reply:
@DearPlanet `vllm.entrypoint.apiserver` is meant as more of a toy example and not being actively improved.

That said I agree if we are using this name for internal purposes it makes more sense to move it into `EngineArgs`, would you be ok making that change? And could augment its existing description with a mention that the first name will also be used in the metrics output. This should also address some of the latest CI test failures.

And then maybe best for the check to be `if not engine_args.served_model_name:` just in case it's an empty list (in theory it shouldn't be).